From: Ben Hutchings <ben@decadent.org.uk>
Date: Fri, 25 Feb 2011 03:23:23 +0000
Subject: [PATCH] Revert "USB: prevent buggy hubs from crashing the USB stack"

This reverts commit 0df14562c291a90c9e74f80b5ee671df474de6cb (commit
d199c96d41d80a567493e12b8e96ea056a1350c1 upstream), included in
longterm 2.6.32.29, which was reported to cause a regression:

http://linux.kernel.org/pipermail/stable-review/2011-February/008336.html
http://linux.kernel.org/pipermail/stable-review/2011-February/008348.html
---
 drivers/usb/core/hub.c |    5 -----
 1 files changed, 0 insertions(+), 5 deletions(-)

diff --git a/drivers/usb/core/hub.c b/drivers/usb/core/hub.c
index 283f019..86768cd 100644
--- a/drivers/usb/core/hub.c
+++ b/drivers/usb/core/hub.c
@@ -2711,11 +2711,6 @@ hub_port_init (struct usb_hub *hub, struct usb_device *udev, int port1,
 		udev->ttport = hdev->ttport;
 	} else if (udev->speed != USB_SPEED_HIGH
 			&& hdev->speed == USB_SPEED_HIGH) {
-		if (!hub->tt.hub) {
-			dev_err(&udev->dev, "parent hub has no TT\n");
-			retval = -EINVAL;
-			goto fail;
-		}
 		udev->tt = &hub->tt;
 		udev->ttport = port1;
 	}
-- 
1.7.4.1

